### PR TITLE
Issue #7734 improvements to charts widgets

### PR DESCRIPTION
**Pull request by ale-cristofori**
## Description
This PR contains changes that took onboard code review suggestions for #7594 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [x] Feature
 - [x] Code style update (formatting, local variables)
 - [x] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**
Within code review of #7594  some issues were raised and enhancements suggested.
1. Replace ColorClassModal classification form input from text box to drop down autocomplete.
2. Bring dropdown menu for classification options in foreground since it was partially hidden by the component on opening.
3. Provide preset (random) colours when users add a new classification rule on ColorClassModal.
4. Include a `cancel` button on the ColorClassModal component asking users for confirmation if some classified values are not completed.
5. Add more margin around the radio button input to chose between `stacked` or `grouped` bar chart type in the Advanced Chart Options section.
6. Solve part of the issue raised with #6217 making more visible the separation lines between slices in pie charts.

**What is the new behavior?**
all changes described above have been applied see [here](https://github.com/geosolutions-it/npa-cgg/issues/214#issuecomment-1010002788) for a complete detailed description of the fixes.

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Review by offtherailz — CHANGES_REQUESTED** [review]
- [ ] I can not see any visible difference between grouped and stacked. image
- [ ] Close and save do exactly the same operation most of the times (no confirm for exit for changes).

https://user-images.githubusercontent.com/1279510/149810553-63a37827-4f43-4c7b-af14-9df5ac7bbe65.mp4


- [ ] close seems to work only on first edit, but if you close before to change something, the interface becomes inconstistent (no colors, no way to add them). 

https://user-images.githubusercontent.com/1279510/149810336-2af140a1-1066-402f-bf3b-2a20f1988a75.mp4
- [ ] I don't remember if for this we decided it was the correct value for `${legendValue}`: image. Anyway it looks a little hard to understand. `legendValue` evokes the value, more than the default label. Maybe two different placeholders is better.

**Comment by tdipisa** [comment]
> * Tooltip should contain, if possible, the labels for each color in the tooltip.

**Comment by ale-cristofori** [comment]
@ale-cristofori @tdipisa @offtherailz, we had an internal meeting where we clarified and discussed the way ahead for the last comments on the ticket.

>  _I can not see any visible difference between grouped and stacked._
This was clarified during the call and is due to how the data is classified in the first place when designing the bar chart parameters. If data cannot be unaggregated any further than the `X Attribute`, which might be the unique identifier for a data set (for states dataset the `STATE_NAME` attribute for example) then there will be no difference between grouped and stacked because data items cannot be grouped or represented as aggregated. That is why in this case we won't be able to see any difference between `stacked` and `grouped`

File: web/client/translations/data.en-US.json [hunk 1]
```diff
@@ -1938,6 +1939,9 @@
               "yAxisLabel": "Legend Label",
               "title": "Advanced Options",
               "legend": "Legend",
+              "barChartType": "Bar Chart Type",
+              "groupedBarChart": "Gruppiert",
+              "stackedBarChart": "Gestapelt",
```
[the remark this answers is not in the document]
Comment (ale-cristofori) [reply web/client/translations/data.en-US.json]: fixed in latest commit

File: web/client/components/style/ThemaClassesEditor.jsx [hunk 2]
```diff
@@ -45,6 +54,23 @@ class ThemaClassesEditor extends React.Component {
...
+                        autocompleteEnabled
+                        column={{key: classificationAttribute}}
+                        onChange={value => this.updateUnique(index, value)}
+                        dataType="string"
+                        typeName={layer.name}
+                        url={ConfigUtils.getParsedUrl(layer.url, {"outputFormat": "json"})}
+                        value={classItem.unique}
+                        filter="contains"
+                        autocompleteStreamFactory={createPagedUniqueAutompleteStream}/>);
+            /** field classes without preset values - text input  */
```
[the remark this answers is not in the document]
Comment (ale-cristofori) [reply web/client/components/style/ThemaClassesEditor.jsx:73]: fixed in the last commit

**Comment by offtherailz** [comment]
@ale-cristofori you should fix missing translation for mandatory languages to make the test run and the check pass

**Review by offtherailz — APPROVED** [review]

**Pull request merged**